open_feature/flag_evaluation/resolution_details.py

```diff
@@ -0,0 +1,17 @@
+import typing
+from dataclasses import dataclass
+
+from open_feature.exception.error_code import ErrorCode
+from open_feature.flag_evaluation.reason import Reason
+
+T = typing.TypeVar("T", covariant=True)
+
+
+@dataclass
+class FlagResolutionDetails(typing.Generic[T]):
+    value: T
+    error_code: typing.Optional[ErrorCode] = None
+    error_message: typing.Optional[str] = None
+    reason: typing.Optional[Reason] = None
+    variant: typing.Optional[str] = None
```

Comment:
I would expect variant to always be defined
```suggestion
    variant: str
```

Reply:
According to the spec, this is optional: https://openfeature.dev/specification/types/#resolution-details